### PR TITLE
Add docs for flatten function

### PR DESCRIPTION
**Pull request by cberner**

### CODE REVIEW

File: presto-docs/src/main/sphinx/functions/array.rst [hunk 1]
```diff
@@ -25,6 +25,10 @@ Array Functions
 
     Remove duplicate values from the array ``x``.
 
+.. function:: flatten(x) -> array
+
+    Flattens an ``array(array(T))`` to an ``array(T)`` by concatenating its elements.
```
Comment (martint) [inline presto-docs/src/main/sphinx/functions/array.rst]: concatenating **their** elements

**Comment by martint** [comment]
looks good

File: presto-docs/src/main/sphinx/functions/array.rst (hunk 1 above)
Comment (cberner) [inline presto-docs/src/main/sphinx/functions/array.rst]: Isn't "its" correct, since I refer to a singular array, "flattens an"?

Comment (cberner) [inline presto-docs/src/main/sphinx/functions/array.rst]: Oh, I see. You were thinking of the sub arrays as the things being concatenated. How about "by concatenating the contained arrays"?

Comment (martint) [inline presto-docs/src/main/sphinx/functions/array.rst]: Sounds good.

**Pull request merged**